SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp

```diff
@@ -0,0 +1,52 @@
+// TODO: enable on Windows once driver is ready
+// REQUIRES: gpu && linux
+// UNSUPPORTED: cuda || hip
+//
+// TODO: enable when Jira ticket resolved
+// XFAIL: gpu
+//
+// Check that full compilation works:
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+/* Tests invoke_simd support in the compiler/headers
+ * Test checks an error message when subgroup size mismatches invoke_simd
+ * return type (simd<T, M> or simd_mask<T, M>) size
+ */
+
+#include "Inputs/common.hpp"
+
+int main(void) {
+  auto q = queue{gpu_selector_v};
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<sycl::info::device::name>()
+            << "\n";
+  bool passed = true;
+
+  // simd_size 8
+  passed &= test<4, 8>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (8) does not match size of invoke_simd return type (4){{.*}}
+  passed &= test<16, 8>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (8) does not match size of invoke_simd return type (16){{.*}}
+  passed &= test<32, 8>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (8) does not match size of invoke_simd return type (32){{.*}}
+
+  // simd_size 16
+  passed &= test<4, 16>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (16) does not match size of invoke_simd return type (4){{.*}}
+  passed &= test<8, 16>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (16) does not match size of invoke_simd return type (8){{.*}}
```

Comment:
Were these originally expected to pass with no error?

Reply:
invoke_simd SPEC was not clear enough.
Actually it requires that If the target of invoke_simd has any simd::experimental::simd<T,N> argument or return, then N must be equal to the sub-group-size in SYCL context.

Reply:
thanks!